userservice/src/main/java/com/infodation/userservice/services/UserService.java

```diff
@@ -0,0 +1,63 @@
+package com.infodation.userservice.services;
+
+import com.infodation.userservice.models.User;
+import com.infodation.userservice.models.dto.user.CreateUserDTO;
+import com.infodation.userservice.models.dto.user.UpdateUserDTO;
+import com.infodation.userservice.repositories.UserRepository;
+import com.infodation.userservice.services.iservice.IUserService;
+import com.infodation.userservice.utils.ApiResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Service
+public class UserService implements IUserService {
+    @Autowired
+    UserRepository userRepository;
```

Comment:
Using `@Autowired ` is a short way but in reality, we prefer using `Constructor Injection` style for projects.
Some reasons:
- Immutability: Dependencies are set once and cannot be changed, making the class more stable.
- Easier to test: It is simpler to create unit tests for classes, as you can easily provide mock dependencies.
- Mandatory dependencies: It makes dependencies clear and enforces them, as Spring will throw an error if the dependencies are not provided.

So please change
```suggestion
    private final UserRepository userRepository;
    public UserServiceImpl(userRepository){ 
     this.userRepository = userRepositor;
    }
```